dbms/src/Encryption/RateLimiter.cpp

```diff
@@ -455,11 +435,10 @@ void IORateLimiter::updateReadLimiter(Int64 bg_bytes, Int64 fg_bytes)
 {
     LOG_FMT_INFO(log, "updateReadLimiter: bg_bytes {} fg_bytes {}", bg_bytes, fg_bytes);
     auto get_bg_read_io_statistic = [&]() {
-        return getCurrentIOInfo().bg_read_bytes;
+        return read_info.bg_read_bytes.load();
     };
     auto get_fg_read_io_statistic = [&]() {
-        auto io_info = getCurrentIOInfo();
-        return std::max(0, io_info.total_read_bytes - io_info.bg_read_bytes);
+        return std::max(0, read_info.total_read_bytes - read_info.bg_read_bytes);
```

Comment:
After `read_info.bg_read_bytes` is updated, a read thread can call `get_fg_read_io_statistic` before `read_info.total_read_bytes` is updated, the return value of `get_fg_read_io_statistic` can be unreasonable.

I suggest we can add a new member `read_info.fg_read_bytes` to avoid this problem.

Also, the default memory order of `std::atomic<T>` is `memory_order_seq_cst`, I think `memory_order_relaxed` is safe in this case. Ref: https://en.cppreference.com/w/cpp/atomic/memory_order

Reply:
maybe we can remove `total_read_bytes` and add `fg_read_bytes`?